File: Graphs - DFS/841. Keys and Rooms/solution.php
class Solution {

    /**
     * @param Integer[][] $rooms
     * @return Boolean
     */
    function canVisitAllRooms($rooms) {
        $visited = array_fill(0, count($rooms), false);
        $visited[0] = true;
        $stack = array(0);
        while (count($stack) > 0) {
            $room = array_pop($stack);
            foreach ($rooms[$room] as $key) {
                if (!$visited[$key]) {
                    $visited[$key] = true;
                    array_push($stack, $key);
                }
            }
        }
        foreach ($visited as $v) {
            if (!$v) return false;
        }
        return true;
    }
}

File: Graphs - DFS/841. Keys and Rooms/solution.cs
public class Solution {
    public bool CanVisitAllRooms(IList<IList<int>> rooms) {
        bool[] visited = new bool[rooms.Count];
        visited[0] = true;
        Stack<int> stack = new Stack<int>();
        stack.Push(0);
        while (stack.Count > 0) {
            int room = stack.Pop();
            foreach (int key in rooms[room]) {
                if (!visited[key]) {
                    visited[key] = true;
                    stack.Push(key);
                }
            }
        }
        foreach (bool v in visited) {
            if (!v) return false;
        }
        return true;
    }
}